src/applications/coronavirus-chatbot/utils.js

```diff
@@ -17,12 +9,35 @@ const disableButtons = event => {
   }
 };
 
-export const watchForButtonClicks = () => {
+const disableCheckboxes = () => {
+  [...document.querySelectorAll('#webchat input[type="checkbox"]')].forEach(
+    input => {
+      const currentInput = input;
+      currentInput.disabled = true;
+      return currentInput;
+    },
+  );
+};
+
+const scrollToNewMessage = () => {
+  const messages = document.getElementsByClassName(
+    'webchat__stackedLayout--fromUser',
+  );
+  const lastMessageFromUser = messages[messages.length - 1];
+  lastMessageFromUser.scrollIntoView({ behavior: 'smooth' });
+};
+
+const handleDisableAndScroll = event => {
+  disableButtons(event);
+  disableCheckboxes();
+  scrollToNewMessage();
+};
+
+export const addEventListenerToButtons = () => {
   setInterval(() => {
     const buttons = document.getElementsByClassName('ac-pushButton');
     for (let i = 0; i < buttons.length; i++) {
-      buttons[i].addEventListener('click', disableButtons);
-      buttons[i].addEventListener('click', scrollToNewMessage);
+      buttons[i].addEventListener('click', handleDisableAndScroll);
```

Comment:
It might be better to use delegated event binding here, i.e. bind 'click' to a wrapping element, then check to see if the correct button was clicked when the event bubbles up.

Reply:
Yes, we had a similar suggestion on a previous [PR](https://github.com/department-of-veterans-affairs/vets-website/pull/12298/files) but found it was messier to do in the `botframework` code.

> The event listener on each button (added by botframework-webchat) doesn't bubble the event up so our event listener on root was never getting executed. It seems messier to change the webchat code, so we stuck with applying our event listener dynamically.